src/call.cpp

```diff
@@ -920,36 +920,69 @@ char * call::get_last_header(const char * name)
         ERROR("call::get_last_header: Header to parse bigger than %d (%zu)", MAX_HEADER_LEN, strlen(name));
     }
 
+    char header_name[MAX_HEADER_LEN];
+    const char *sep = strrchr(name, ':');
+
+    if (!sep){ /* [last_Header] */
+        sprintf(header_name, "%s:", name);
+        return get_header(last_recv_msg, header_name, false);
+    }
+
+    snprintf(header_name, len - strlen(sep) + 2, name);
+
+    /* [last_Header:] */
     if (name[len - 1] == ':') {
         return get_header(last_recv_msg, name, false);
+    }
+    char *value = get_header(last_recv_msg, header_name, true);
+
+    if (!strcmp(sep, ":value")) { /* [last_Header:value] */
+        ; /* done */
+    } else if (!strcmp(sep, ":uri")) { /* [last_Header:uri] */
+        char * uri = get_last_header_uri(header_name);
+        int uri_len = strlen(uri);
+        memmove(value, uri, uri_len);
+        value[uri_len] = '\0';
+        free(uri);
```

Comment:
If get_last_header_uri is dropped, we get rid of this unusual free() here (because of the different semantics of get_last_header_uri).